vendor/zkllvm-metacraft-circuits/src/circuits_imp/compute_shuffled_index_imp.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <nil/crypto3/hash/algorithm/hash.hpp>
+#include <nil/crypto3/hash/sha2.hpp>
+
+#include <algorithm>
+#include <array>
+
+#include "compute_shuffled_index_imp.h"
+#include "../circuit_utils/circuit_byte_utils.h"
+#include "../utils/picosha2.h"
+
+using namespace circuit_byte_utils;
+
+uint64_t compute_shuffled_index_imp(uint64_t index, uint64_t index_count, std::array<Byte, 32> seed,
+                                            int SHUFFLE_ROUND_COUNT = 90) {
+    assert_true(index < index_count);
+
+    std::array<Byte, 32 + 1 + 4> source_buffer {};
+
+    //!!! sha256_to_bytes_array(seed, source_buffer);
+    std::copy(seed.begin(), seed.end(), source_buffer.begin());
```

Comment:
I was left under the impression that std::copy doesn't work and you had to implement it in a dumb way. Are you sure that this compiles as a circuit?